main/php_syslog.c

```diff
@@ -32,6 +32,46 @@
 #define syslog std_syslog
 #endif
 
+PHPAPI void php_syslog_str(int priority, const zend_string* message)
+{
+	smart_string sbuf = {0};
+
+	if (PG(syslog_filter) == PHP_SYSLOG_FILTER_RAW) {
+		/* Just send it directly to the syslog */
+		syslog(priority, "%s", ZSTR_VAL(message));
+		return;
+	}
+
+	/* We use < because we don't want the final NUL byte to be converted to '\x00' */
+	for (size_t i = 0; i < ZSTR_LEN(message); ++i) {
+		unsigned char c = ZSTR_VAL(message)[i];
+
+		/* check for NVT ASCII only unless test disabled */
+		if (((0x20 <= c) && (c <= 0x7e)))
+			smart_string_appendc(&sbuf, c);
+		else if ((c >= 0x80) && (PG(syslog_filter) != PHP_SYSLOG_FILTER_ASCII))
+			smart_string_appendc(&sbuf, c);
+		else if (c == '\n') {
+			/* Smart string is not NUL terminated */
+			syslog(priority, "%.*s", (int)sbuf.len, sbuf.c);
+			smart_string_reset(&sbuf);
+		} else if ((c < 0x20) && (PG(syslog_filter) == PHP_SYSLOG_FILTER_ALL))
+			smart_string_appendc(&sbuf, c);
+		else {
+			const char xdigits[] = "0123456789abcdef";
+
+			smart_string_appendl(&sbuf, "\\x", 2);
+			smart_string_appendc(&sbuf, xdigits[(c / 0x10)]);
+			c &= 0x0f;
+			smart_string_appendc(&sbuf, xdigits[c]);
```

Comment:
```suggestion
			smart_string_appendc(&sbuf, xdigits[c >> 4]);
			smart_string_appendc(&sbuf, xdigits[c & 0xf]);
```
Found the style a bit odd here.